src/main/java/io/openliberty/tools/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -288,6 +304,21 @@ private File loadInstallJarFile(File installDirectory) {
      */
     public abstract File downloadArtifact(String groupId, String artifactId, String type, String version)
             throws PluginExecutionException;
+    /**
+     * Download the signature file from the specified Maven coordinates, or retrieve it
+     * from the cache if it already exists. This is same as downloadArtifact method, but for Gradle 
+     * the signature file need to be copied to ESA file directory to be verified. 
+     * 
+     * @param esa    	 The ESA file for the signature
+     * @param groupId    The group ID
+     * @param artifactId The artifact ID
+     * @param type       The type e.g. esa
+     * @param version    The version
+     * @return The file corresponding to the downloaded artifact
+     * @throws PluginExecutionException If the artifact could not be downloaded
+     */
+    public abstract File downloadSignature(File esa, String groupId, String artifactId, String type, String version) 
```

Comment:
This new abstract method will require the changes to maven/gradle. So ignore my other comment about the constructor.